plugins/processors/awsappsignals/processor_test.go

```diff
@@ -0,0 +1,239 @@
+	expoHistogram := m.SetEmptyExponentialHistogram().DataPoints().AppendEmpty()
+	expoHistogram.SetSum(10)
+
+	m = md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
+	summary := m.SetEmptySummary().DataPoints().AppendEmpty()
+	summary.SetSum(10)
+
+	m = md.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty()
+	histogram := m.SetEmptyHistogram().DataPoints().AppendEmpty()
+	histogram.SetSum(10)
+
+	for k, v := range dimensions {
+		gauge.Attributes().PutStr(k, v)
+		sum.Attributes().PutStr(k, v)
+		expoHistogram.Attributes().PutStr(k, v)
+		summary.Attributes().PutStr(k, v)
+		histogram.Attributes().PutStr(k, v)
+	}
+
+	return md
+}
+
+func getDimensionValue(t *testing.T, m pmetric.Metrics, dimensionName string) string {
+	var agreedValue string
+
+	gauge := m.ResourceMetrics().At(0).ScopeMetrics().At(0).Metrics().At(0).Gauge()
+	if val, ok := gauge.DataPoints().At(0).Attributes().Get(dimensionName); !ok {
+		t.Errorf("no dimension value is found with key %s\n", dimensionName)
+	} else {
+		agreedValue = val.AsString()
+	}
+
+	sum := m.ResourceMetrics().At(1).ScopeMetrics().At(0).Metrics().At(0).Sum()
+	if val, ok := sum.DataPoints().At(0).Attributes().Get(dimensionName); !ok {
+		t.Errorf("no dimension value is found with key %s\n", dimensionName)
+	} else {
+		newVal := val.AsString()
+		if agreedValue != newVal {
+			t.Errorf("inconsistent dimension value, agreed value is %s, new %s\n", agreedValue, newVal)
+		}
+	}
+
+	expoHistogram := m.ResourceMetrics().At(2).ScopeMetrics().At(0).Metrics().At(0).ExponentialHistogram()
+	if val, ok := expoHistogram.DataPoints().At(0).Attributes().Get(dimensionName); !ok {
+		t.Errorf("no dimension value is found with key %s\n", dimensionName)
+	} else {
+		newVal := val.AsString()
+		if agreedValue != newVal {
+			t.Errorf("inconsistent dimension value, agreed value is %s, new %s\n", agreedValue, newVal)
+		}
+	}
+
+	summary := m.ResourceMetrics().At(3).ScopeMetrics().At(0).Metrics().At(0).Summary()
+	if val, ok := summary.DataPoints().At(0).Attributes().Get(dimensionName); !ok {
+		t.Errorf("no dimension value is found with key %s\n", dimensionName)
+	} else {
+		newVal := val.AsString()
+		if agreedValue != newVal {
+			t.Errorf("inconsistent dimension value, agreed value is %s, new %s\n", agreedValue, newVal)
+		}
+	}
+
+	histogram := m.ResourceMetrics().At(4).ScopeMetrics().At(0).Metrics().At(0).Histogram()
+	if val, ok := histogram.DataPoints().At(0).Attributes().Get(dimensionName); !ok {
+		t.Errorf("no dimension value is found with key %s\n", dimensionName)
+	} else {
+		newVal := val.AsString()
+		if agreedValue != newVal {
+			t.Errorf("inconsistent dimension value, agreed value is %s, new %s\n", agreedValue, newVal)
+		}
+	}
+	return agreedValue
+}
+
+func isMetricNil(m pmetric.Metrics) bool {
+	dps := m.ResourceMetrics().At(0).ScopeMetrics().At(0).Metrics().At(0).Gauge().DataPoints()
+	if dps.Len() > 0 {
```

Comment:
wont dps.Len() be the same for all of the if conditions since we are never re-assigning `dps`?

Reply:
Thanks for pointing it out. Updated.